source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -69,7 +69,9 @@ void MetricsServiceSink::flushCounter(const Stats::Counter& counter) {
   metrics_family->set_type(io::prometheus::client::MetricType::COUNTER);
   metrics_family->set_name(counter.name());
   auto* metric = metrics_family->add_metric();
-  metric->set_timestamp_ms(std::chrono::system_clock::now().time_since_epoch().count());
+  metric->set_timestamp_ms(std::chrono::duration_cast<std::chrono::milliseconds>(
+                               std::chrono::system_clock::now().time_since_epoch())
+                               .count());
```

Comment:
Looks great; can you factor out this verbose function call -- repeated 3x, into a helper function.

Reply:
Ok, I can put the help function on the anonymous namespace in the current file?Or in the common/common/utility?

Reply:
easiest to put it in local anon namespace, I will be moving it into TimeSource anyway.

There's a bit of debt here -- we should be using TimeSource::systemTime() rather than directly reading the system-clock, so that this code can have time-injection, but I'll fix that after you merge.

Reply:
And actually, can you add (near your helper method):
  // TODO(#4160): use TimeSource::systemTime() rather than directly reading the system-clock

Reply:
OK